First version of community guidelines with Help Wanted process

jasonblais opened the pull request:
This is a first version of community guidelines, review and feedback welcome.

Addresses: 
 - GitHub process for the Platform repo (to be extended to further repos in the future)
 - Community forum process, largely copied from an internal document

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+ - a link to the `developer’s guide <https://docs.mattermost.com/guides/developer.html>`_ for `machine setup <https://docs.mattermost.com/developer/developer-setup.html>`_ and `developer workflow <https://docs.mattermost.com/developer/developer-flow.html>`_
+ - a link to the `code contribution guidelines <https://docs.mattermost.com/developer/contribution-guide.html>`_
+    
+- has a well-defined difficulty level, both in the description and with the corresponding label
```
Does this mean adding **Difficulty Level**: Introductory to the descriptions? I'm wondering if there's a way to make it more clear by using a template for what the Help Wanted issue should look like

jasonblais replied:
Yeah, I'm actually starting to think that a visual would be better to describe the different pieces, followed by a template.

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+Managing help wanted issues
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once a help wanted issue is added, the creator is responsible for
```
add a colon here?

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+Goal
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Increase Mattermost usage through more numerous and larger deployments via vibrant open source community by contributing features that would not otherwise be offered
```
Maybe change phrasing to something like: 

Increase Mattermost usage through more numerous and larger deployments via a vibrant open source community that contributes features that would not otherwise be offered.

yangchen1 replied:
Add period at end? Also "larger and more numerous deployments" flows better to me, it doesn't imply "more larger" that way. Also, if we're going for full sentence here, it should say via *a* vibrant... Also, should the "by" be there? It not, there should be a comma after "community"

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+
+- has a link to the corresponding JIRA ticket 
+
+- gets noted in the corresponding JIRA ticket
```
maybe change to gets added to the Jira ticket description, because I think devs were sometimes missing it in the comments section

yangchen1 commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+Principles
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Contributors get a worthwhile benefit from their contributions from the feature directly, and/or from social recognition
```
contributions, from the feature directly and/or from social recognition

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+
+----
+
+Community Forums
```
This looks good, I'm wondering about the best way to fit some of this stuff together - I have the doc with the overall Community PM responsibilities which semi-overlaps with the response timeline portion, and IT has a communication doc that semi-overlaps with the response writing tips section.  

Not sure if we should somehow link between them, or if the duplication is fine for now.

yangchen1 commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Contributors get a worthwhile benefit from their contributions from the feature directly, and/or from social recognition
+- Help Wanted issues should have a low barrier to contribute
```
Is it possible to re-word this a bit? I know what this is trying to say, but the metaphor is not totally clear to me.

jasonblais replied:
Would you have any suggestions for an alternative wording?

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- After responding to an issue, if the asker comes back and lets you know that the issue is solved, please add [Solved] to the beginning of the title of the question
+  - Don't be afraid to close GitHub issues if you think you've solved them
```
Is this covering GitHub Issues, forums, or both? The top of the doc says this is about forums, and I'm not sure that we put solved in the GitHub issue title

jasonblais replied:
Just for forums to start with, took it out,

yangchen1 commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+
+When creating a help wanted issue, please ensure it contains the following (`see a sample help wanted issue here <https://github.com/mattermost/platform/issues/4755>`_):
+
+- is well-defined and unambiguous
```
"ensure it contains the following" does make sense with "is well-defined and unambiguous". Maybe either say "Well-defined and unambiguous instructions" or change "please ensure it contain the following" to something that fits with the list

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,141 @@
+
+In this document we set out some easy-to-follow community guidelines, including:
+
+- `GitHub Help Wanted Issues in Platform Repository`_ - Guidelines for managing Help Wanted issues in GitHub
```
Is there any way we could shorten / make the formatting look a little better here? Not a big deal if we can't, the first line is just a little long

<img width="687" alt="screenshot 2016-12-15 14 18 53" src="https://cloud.githubusercontent.com/assets/9354841/21238492/46a94f62-c2d1-11e6-811c-df3cd1bbab71.png">

jasonblais commented:
Thanks for the feedback! Most of it addressed!

I'm going to add a WIP label for this however and get back to it after holidays. I think for the Help Wanted issue, I'm going to change it to a visual followed by a template rather than text to make it easier to follow and understand.

Maybe at that time we can also figure out whether to combine communications docs (from IT), community PM responsibilities (from LB) and the community guidelines from here (or whether they should be separate docs)

jasonblais commented:
Updated, just waiting for one more discussion with IT later this week.

jasonblais commented:
Ready for PM review

it33 commented:
@grundleborg per discussion on community at offsite yesterday, wondering if you'd have feedback on this document?

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,127 @@
+
+2 - After creating the ticket, add "(Proposed APR)" to its title, so the triage team knows to consider it for a ``help-wanted`` label.
+
+3 - If the Jira ticket is accepted as a Help Wanted issue during `triage <https://docs.mattermost.com/process/training.html#triage-meeting>`_, a ``help-wanted`` label is added. This action automatically triggers a zap that creates a new issue in the `platform repository <https://github.com/mattermost/platform>`_.
```
Should we add a note about what we should look for in triage to accept a Help Wanted? (Also should we assign difficulty level/programming language then? Or afterwards?)

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,127 @@
+Once a Help Wanted issue is added, the PM on community rotation is responsible for:
+
+- answering questions from community members, which get automatically posted in the `Community Heartbeat channel <https://pre-release.mattermost.com/core/channels/community-heartbeat>`_
+- following up with a community member if there hasn't been a response for more than two weeks
```
Should this be a part of the community rotation PM? Or should there be a PM assigned to the ticket or something like that instead

jasonblais replied:
There's a whole bunch of smaller tickets like bug fixes as well, not sure if it's worth assigning a PM for those + PMs probably wouldn't remember which ticket was assigned to whom.

The PM on community rotation can ping relevant PMs as needed.

lfbrock commented on source/process/community-guidelines.rst:
```diff
@@ -0,0 +1,127 @@
+- Elevate (thank, recognize and approve their work)
+- Grow (invite people to help as experts, promote people)
+
+Response timeline
```
This section might semi-overlap with https://docs.mattermost.com/process/community-process.html#community-channels

jasonblais replied:
I shortened it to focus on response guidelines

jasonblais commented:
Updated:
 - addressed feedback from @lfbrock 
 - updated docs to change references of "APR" to "Help Wanted"

Note: As one of the planned community improvements, we'll be cleaning up some of the developer/community docs since they're a little all over the place right now.

lfbrock commented:
+1
